### ISSUE TITLE
Migrate to TensorFlow 2.X

### ISSUE THREAD

**Issue by TjarkMiener** [ctlearn-project/ctlearn#127]
I was trying to update the CTLearn 'multitask' branch to TensorFlow 2.X and I faced some difficulties/concerns about the HeadAPI and Graph structure with Estimators, which I wanted to discuss. I pushed some (unfinished) code in the 'tf2' branch here in CTLearn. So far, I focused on the single_tel model. The code starts training (with reasonable loss values), but don't increase the steps (-> train forever without reaching the evaluation). I'm not sure yet what's causing this behavior. 

What is the problem about the HeadAPI and Graph structure in TF 2.X?

As you all know, TensorFlow 2.X integrated the Keras API to make the framework more pythonic. Essentially, moving from the Graph structure with Estimators to the eager execution. Since the TF layers are deprecated, TF 2.X are forcing the users to use the Keras Layers and the Keras Model. However, the estimators and Graph structure are still "supported". It might be better to say they are still there than supported. [This blog post](https://pgaleone.eu/tensorflow/keras/2019/01/19/keras-not-yet-interface-to-tensorflow/) sum up the situation pretty well (worth reading the conclusion at least). This post is pointing out that the Keras Models (with Keras Layers) are not interacting very nicely with a custom non-Keras way of calculating and minimizing the loss (as the HeadAPI does).

Besides that the current 'tf2' branch printing out some warnings that function will be deprecated in the future. 
1) Due to our Graph structure: 
```
WARNING:tensorflow:From /home/tjark/anaconda3/envs/tf2/lib/python3.7/site-packages/tensorflow_core/python/training/training_util.py:236: Variable.initialized_value (from tensorflow.python.ops.variables) is deprecated and will be removed in a future version.
Instructions for updating:
Use Variable.read_value. Variables in 2.X are initialized automatically both in eager and graph (inside tf.defun) contexts.
WARNING:From /home/tjark/anaconda3/envs/tf2/lib/python3.7/site-packages/tensorflow_core/python/training/training_util.py:236: Variable.initialized_value (from tensorflow.python.ops.variables) is deprecated and will be removed in a future version.
Instructions for updating:
Use Variable.read_value. Variables in 2.X are initialized automatically both in eager and graph (inside tf.defun) contexts.
WARNING:tensorflow:From ctlearn/run_model.py:240: DatasetV1.make_one_shot_iterator (from tensorflow.python.data.ops.dataset_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use 'for ... in dataset:' to iterate over a dataset. If using 'tf.estimator', return the 'Dataset' object directly from your input function. As a last resort, you can use 'tf.compat.v1.data.make_one_shot_iterator(dataset)'.
```
2) Due to the HeadAPI implementation:
```
WARNING:tensorflow:From /home/tjark/anaconda3/envs/tf2/lib/python3.7/site-packages/tensorflow_estimator/python/estimator/head/binary_class_head.py:206: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.cast` instead.
```

Multitask Learning is also supported by the [Keras API](https://keras.io/getting-started/functional-api-guide/), so we could potentially switch to Keras, when updating to TF 2.X. Any thoughts?

**Comment by TjarkMiener** [comment]
Hi @STSpencer! Thanks for reactivating this important issue! Given the fact that most of the newest GPUs do not support CUDA10 we are forced to migrate the keras backend of TF2.X.  This issue is the most pressing one.
